src/frontend/src/lib/services/request-pouh-credential.services.ts

```diff
@@ -11,31 +11,79 @@ import { i18n } from '$lib/stores/i18n.store';
 import { userProfileStore } from '$lib/stores/settings.store';
 import { toastsError } from '$lib/stores/toasts.store';
 import { popupCenter } from '$lib/utils/window.utils';
+import type { Identity } from '@dfinity/agent';
 import { Principal } from '@dfinity/principal';
-import { isNullish, nonNullish } from '@dfinity/utils';
+import { fromNullable, isNullish, nonNullish } from '@dfinity/utils';
 import {
 	requestVerifiablePresentation,
 	type VerifiablePresentationResponse
 } from '@dfinity/verifiable-credentials/request-verifiable-presentation';
 import { get } from 'svelte/store';
+import { loadCertifiedUserProfile } from './load-user-profile.services';
 
-const handleSuccess = async (
-	response: VerifiablePresentationResponse
-): Promise<{ success: boolean }> => {
+const addPouhCredential = async ({
+	identity,
+	credentialJwt,
+	issuerCanisterId
+}: {
+	identity: Identity;
+	credentialJwt: string;
+	issuerCanisterId: Principal;
+}): Promise<{ success: boolean }> => {
+	const { auth: authI18n } = get(i18n);
+	try {
+		const userProfile = get(userProfileStore);
+		const response = await addUserCredential({
+			identity,
+			credentialJwt,
+			credentialSpec: {
+				credential_type: POUH_CREDENTIAL_TYPE,
+				arguments: []
+			},
+			issuerCanisterId,
+			currentUserVersion: fromNullable(userProfile?.version ?? [])
+		});
+		if ('Ok' in response) {
+			return { success: true };
+		}
+		if ('Err' in response) {
+			if ('InvalidCredential' in response.Err) {
+				toastsError({
+					msg: { text: authI18n.error.invalid_pouh_credential }
+				});
+				return { success: false };
+			}
+			// Throw so that it gets handled by the catch block.
+			throw new Error();
```

Comment:
the linter or the editor does not display a hint here? something like throwing within a try catch is not recommended. Can you return an error?

Reply:
I want the same behavior as inside the `catch` block.

There are three options:

* Move the code of the `catch` after the `try/catch` block and do nothing here nor inside the `catch` block.
* Duplicated the toast error here instead of throwing.
* Throw an error.

Which one do you prefer?

Reply:
I would have two different errors since they are distinct. I would create a specific label for the first toast to make it obvious that it throws an error because the response type is not supported.

```
try {
     toastsError({
			msg: { text: authI18n.error.a_new_label_for_response_type_unsupported },
		});

} catch (....
```